sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/tracing/TraceTokenClientInterceptor.java

```diff
@@ -12,26 +12,33 @@
 import feign.RequestInterceptor;
 import feign.RequestTemplate;
 import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.web.context.request.RequestAttributes;
 import org.springframework.web.context.request.RequestContextHolder;
 
 public class TraceTokenClientInterceptor implements RequestInterceptor {
 
+  private static final Logger LOG = LoggerFactory.getLogger(TraceTokenClientInterceptor.class);
+
   @Override
   public void apply(RequestTemplate template) {
     var traceToken = computeTraceTokenIfAbsent();
     template.header(TRACE_TOKEN_HEADER_NAME, traceToken);
   }
 
   private String computeTraceTokenIfAbsent() {
-    var attribute =
+    try {
+      var attribute =
+          RequestContextHolder.currentRequestAttributes()
+              .getAttribute(TRACE_TOKEN_HEADER_NAME, RequestAttributes.SCOPE_REQUEST);
+      if (attribute instanceof String strAttribute) {
         RequestContextHolder.currentRequestAttributes()
-            .getAttribute(TRACE_TOKEN_HEADER_NAME, RequestAttributes.SCOPE_REQUEST);
-
-    if (attribute instanceof String strAttribute) {
-      RequestContextHolder.currentRequestAttributes()
-          .setAttribute(TRACE_TOKEN_HEADER_NAME, strAttribute, RequestAttributes.SCOPE_REQUEST);
-      return strAttribute;
+            .setAttribute(TRACE_TOKEN_HEADER_NAME, strAttribute, RequestAttributes.SCOPE_REQUEST);
+        return strAttribute;
+      }
+    } catch (IllegalStateException e) {
```

Comment:
It's unfortunate that there is no clean way to find out if we're in a request context instead of catching the exception.

As alternative we could invoke `RequestContextHolder#getRequestAttributes` which would return `null` if the attributes are missing. Still it's an indirect way to find out and the implementation is slighty different because it does not fall back to `FacesRequestAttributesFactory` in case of JSF.

Reply:
Just from the wording, without knowing the actual implementation of the foreign library, _current_ sounds much more reasonable than _get_. What's so bad about catching a documented exception?

Reply:
Just that it should be an "exception" and not a regular use case :D 
But I'm fine with your approach since there is no better alternative like `hasRequestContext`.

Reply:
Well, first checking and then accessing can lead to edge cases. Maybe not for something that is bound to the current thread like here. But in many other cases, checking first and then accessing is not a good idea. Examples: Accessing existing or non-existing files after checking _exists_, modifying data in a non-transactional database after checking if it matches a specific criteria. There could always be another process modifying it at the same time.